src/controller/ensMetadata.ts

```diff
@@ -0,0 +1,68 @@
+import { Request, Response } from 'express';
+import { FetchError } from 'node-fetch';
+import { ContractMismatchError, UnsupportedNetwork } from '../base';
+import { checkContract } from '../service/contract';
+import { getDomain } from '../service/domain';
+import getNetwork from '../service/network';
+import { getLabelhash } from '../utils/labelhash';
+
+export async function ensMetadata(req: Request, res: Response) {
+  // #swagger.description = 'ENS NFT metadata'
+  // #swagger.parameters['networkName'] = { schema: { $ref: '#/definitions/networkName' } }
+  // #swagger.parameters['{}'] = { name: 'contractAddress', description: 'Contract address which stores the NFT indicated by the tokenId', schema: { $ref: '#/definitions/contractAddress' } }
+  // #swagger.parameters['tokenId'] = { type: 'string', description: 'Namehash(v1) /Labelhash(v2) of your ENS name.\n\nMore: https://docs.ens.domains/contract-api-reference/name-processing#hashing-names', schema: { $ref: '#/definitions/tokenId' } }
+  const { contractAddress, networkName, tokenId } = req.params;
+
+  // check if token id provided as raw ens name, if so then convert to labelhash
+  // TODO add namehash conversion for v2
+  let _tokenId;
+  if (tokenId.endsWith('.eth')) {
```

Comment:
Seems identical to https://github.com/ensdomains/ens-metadata-service/pull/77/files#diff-9e6ab3c59dcced912c117bc61f7e34c3a6c3e225e4d2586f5366f62916756bb5R20 Maybe refactor into one sharable code?